course_gather/models.py

```diff
@@ -88,18 +126,55 @@ class MTUCourse(models.Model):
+    mtu_course_crn = models.CharField(
                             max_length=50,
-                            help_text='The id of the MTU course')
-    mtu_course_name = models.CharField(
-                            max_length=50,
-                            help_text='The Name of the MTU course')
-    mtu_subject = models.CharField(
+                            help_text='The CRN of the MTU course')
+    mtu_course_subject = models.CharField(
                         max_length=50,
                         help_text='The subject of the MTU course')
-    mtu_credits = models.FloatField(
+    mtu_course_id = models.CharField(
+                        max_length=10,
+                        help_text='The course ID of the MTU course')
+    mtu_course_section = models.CharField(
+                            max_length=3,
+                            help_text='The section of the MTU course')
+    mtu_course_campus = models.CharField(
+                            max_length=2,
+                            default="1",
+                            help_text='The campus code of the MTU course')
+    mtu_course_credits = models.FloatField(
                         blank=True,
                         help_text='The amount of credits the MTU course is')
+    mtu_course_name = models.CharField(
+                            max_length=50,
+                            help_text='The Name of the MTU course')
+    mtu_course_days = models.CharField(
+                            max_length=7,
+                            help_text='The days the class is scheduled for')
+    mtu_course_time = models.CharField(
+                            max_length=40,
+                            help_text='The hours the MTU course runs')
+    mtu_course_capacity = models.IntegerField(
+                            help_text='The capacity of the class')
+    mtu_course_capacity_enrolled = models.IntegerField(
+                            help_text='The amount of students currently \
+                            enrolled in the course')
+    mtu_course_capacity_remaining = models.IntegerField(
+                            help_text='The seats remaining in the course')
+    mtu_course_instructor = models.CharField(
+                            max_length=60,
+                            help_text='The last name of the professor(s) \
+                            teaching the course')
+    mtu_course_duration = models.CharField(
+                            max_length=30,
+                            help_text='The date range the class runs')
+    mtu_course_location = models.CharField(
+                            max_length=10,
+                            help_text='The MTU building location code the \
+                            class is located in')
+    mtu_course_fee = models.CharField(
+                            max_length=80,
+                            help_text='The fee associated with the course')
 
     def __str__(self):
         return self.mtu_course_id
```

Comment:
Are we still planning to use a composite primary key?

Reply:
Yep, we're just going to need that data from the scraper first

> I'm having some trouble understanding the Shell model. Why is it a many-to-many relation between Shell and MTUCourse?

TL;DR, there can be 0, 1, or many classes with a given name, and there can be 0, 1, or many transfer classes that can apply to the MTU classes.

The biggest reason for this is that there are often times multiple sections associated with an MTU course CRN. For example, Physics By Inquiry II has 17 sections. This Shell model is meant to represent the name of the course so we can relate those 17 MTU classes to the 89 transfer courses that are in the current dataset.

Reply:
Got it.

In that case, would it make sense to omit the `name` field from MTUCourse since it's duplicated in Shell instances?

Reply:
Not really because we're probably going to want that data for the front end, also imo it makes sense to keep the duplicated data in the object. This object structure is more or less just for easy searching UI/API wise. But if we omit the name instance, while we're in our loops we lose some context unless we go out of the way to track the top level

Reply:
That said, later on once we actually get data and start work on the UI and get that working, if we want to remove it, it's a simple fix :man_shrugging: .

Reply:
👍